java/google/registry/whois/WhoisCommandFactory.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 The Nomulus Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google.registry.whois;
+
+import com.google.common.net.InternetDomainName;
+import google.registry.config.RegistryConfig.ConfigModule;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+
+/**
+ * A no-op base class used to provide custom whois commands.
+ * <p>To add custom commands, extend this class, then configure it in
+ * {@link ConfigModule#provideWhoisCommandFactoryClass()}.
+ */
+public class WhoisCommandFactory {
+
+  public WhoisCommand domainLookupCommand(InternetDomainName domainName) {
+    return domainLookupCommand(domainName, null);
+  }
+
+  public WhoisCommand domainLookupCommand(InternetDomainName domainName, @Nullable InternetDomainName tld) {
+    return new DomainLookupCommand(domainName, tld);
+  }
+
+  public WhoisCommand nameserverLookupByIpCommand(InetAddress inetAddress) {
+    return new NameserverLookupByIpCommand(inetAddress);
+  }
+
+  public WhoisCommand nameserverLookupByHostCommand(InternetDomainName hostName) {
```

Comment:
Mark this final -- it shouldn't be overridable since it's just delegating to another method that can be overridden that can handle all possible cases.